cvat-cli/src/cvat_cli/_internal/commands_functions.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (C) 2024 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import argparse
+import json
+import textwrap
+from collections.abc import Sequence
+
+import cvat_sdk.auto_annotation as cvataa
+from cvat_sdk import Client
+
+from .agent import run_agent
+from .command_base import CommandGroup
+from .common import FunctionLoader, configure_function_implementation_arguments
+
+COMMANDS = CommandGroup(description="Perform operations on CVAT lambda functions.")
+
+
+@COMMANDS.command_class("create-native")
+class FunctionCreateNative:
+    description = textwrap.dedent(
+        """\
+        Create a CVAT function that can be powered by an agent running the given local function.
+        """
+    )
+
+    def configure_parser(self, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            "name",
+            help="a human-readable name for the function",
+        )
+
+        configure_function_implementation_arguments(parser)
+
+    def execute(
+        self,
+        client: Client,
+        *,
+        name: str,
+        function_loader: FunctionLoader,
+    ) -> None:
+        function = function_loader.load()
+
+        remote_function = {
+            "provider": "native",
+            "name": name,
+        }
+
+        if isinstance(function.spec, cvataa.DetectionFunctionSpec):
+            remote_function["kind"] = "detector"
+            remote_function["labels_v2"] = []
+
+            for label_spec in function.spec.labels:
+                if getattr(label_spec, "sublabels", None):
+                    raise cvataa.BadFunctionError(
+                        f"Function label {label_spec.name!r} has sublabels. This is currently not supported."
+                    )
+
+                remote_function["labels_v2"].append(
+                    {
+                        "name": label_spec.name,
+                    }
+                )
+        else:
+            raise cvataa.BadFunctionError(
+                f"Unsupported function spec type: {type(function.spec).__name__}"
+            )
+
+        _, response = client.api_client.call_api(
+            "/api/functions",
+            "POST",
+            body=remote_function,
+        )
```

Comment:
Do you think if SDK-level functions for the functions API could be useful?

Reply:
In principle, yes, but I'm not going to implement that in this PR. For now, I consider everything here to be a CLI-only feature.

Ideally, I'd prefer to add some stubs for the enterprise function API to the server, so that the normal SDK auto-generation can work. But that'll have to be another time.

Reply:
Usually, "CLI-on;y" features result in unnecessary obstacles when you're trying to customize the CLI behavior in a custom script, so adding at least something at the SDK level would be a great extensibility feature. I'm not insisting on doing it in this PR.